beard_server/modules/text/views.py

```diff
@@ -51,35 +51,61 @@
 def available_methods():
     """List of available methods to call as the part of API."""
     available = {
-        "clusters": {
-            "path": "/api/text/phonetic_block",
-            "method": "[GET]",
+        "phonetic_blocks": {
+            "path": "/api/text/phonetic_blocks",
+            "method": "[POST]",
             "example": {
-                "request": "/api/text/phonetic_block?full_name=John%20Smith",
-                "response": {"phonetic_block": "SNATHj"}
+                "body": {
+                    "full_names": [
+                        "John Smith"
+                    ]
+                },
+                "header": {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json"
+                },
+                "response": {
+                    "phonetic_blocks": {
+                        "John Smith": "SNATHj"
+                    }
+                }
             }
         }
     }
 
     return jsonify(available)
 
 
-@blueprint.route('/phonetic_block', methods=['GET'])
-def get_phonetic_block():
+@blueprint.route('/phonetic_blocks', methods=['POST'])
+def get_phonetic_blocks():
     """Create phonetic block from a given full name using nysiis algorithm."""
+    full_names_data = request.get_json(silent=True)
+    response = {'phonetic_blocks': []}
+
+    # Unpack the body of the POST request.
     try:
-        name = {'author_name': request.values.get('full_name', type=unicode)}
+        full_names = []
+
+        for full_name in full_names_data['full_names']:
+            try:
+                full_names.append({'author_name': unicode(full_name)})
+            except TypeError:
+                # Malformed author's name.
+                response['phonetic_blocks'].append({full_name: ""})
+    except KeyError:
+        # Missing data.
+        abort(400)
 
-        signature_block = block_phonetic(
-            np.array([name], dtype=np.object).reshape(-1, 1),
+    signature_blocks = list(
+        block_phonetic(np.array(
+            full_names,
+            dtype=np.object).reshape(-1, 1),
             threshold=0,
             phonetic_algorithm='nysiis'
-        )[0]
-    except IndexError:
-        # Most likely a malformed author name.
-        abort(420)
-    except TypeError:
-        # Missing full_name argument.
-        abort(400)
+        )
+    )
+
```

Comment:
Please, move this black magic  into a separate function that is used in both bulk and single calls.